imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -47,47 +41,50 @@
 public class CdrFileServiceImpl implements CdrFileService {
 
     private static final String CDR_FILE_DIRECTORY = "imi.cdr_file_directory";
-    private static final String RESCHEDULE_CALL = "nms.imi.reschedule_call";
-    private static final String DEACTIVATE_SUBSCRIPTION = "nms.imi.deactivate_subscription";
-    private static final String COMPLETE_SUBSCRIPTION = "nms.imi.complete_subscription";
+    private static final String PROCESS_CDR = "nms.imi.kk.process_cdr";
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CdrFileServiceImpl.class);
     public static final String CDR_SUMMARY = "cdrSummary";
 
     private SettingsFacade settingsFacade;
     private EventRelay eventRelay;
-    private AuditDataService auditDataService;
+    private FileAuditRecordDataService fileAuditRecordDataService;
     private AlertService alertService;
-    private CallDetailRecordDataService cdrDataService;
-    private SubscriptionService subscriptionService;
 
 
 
     @Autowired
     public CdrFileServiceImpl(@Qualifier("imiSettings") SettingsFacade settingsFacade, EventRelay eventRelay,
-                              AuditDataService auditDataService, AlertService alertService,
-                              CallDetailRecordDataService cdrDataService,
-                              SubscriptionService subscriptionService) {
+                              FileAuditRecordDataService fileAuditRecordDataService, AlertService alertService) {
         this.settingsFacade = settingsFacade;
         this.eventRelay = eventRelay;
-        this.auditDataService = auditDataService;
+        this.fileAuditRecordDataService = fileAuditRecordDataService;
         this.alertService = alertService;
-        this.cdrDataService = cdrDataService;
-        this.subscriptionService = subscriptionService;
     }
 
 
-    private List<CallDetailRecord> readCdrs(String cdrFileLocation, String summaryFileName, String checksum) {
+    private void reportErrorAndThrow(String file, String error) {
+        LOGGER.error(error);
+        alertService.create(file, CDR_SUMMARY, error, AlertType.CRITICAL, AlertStatus.NEW, 0, null);
+        fileAuditRecordDataService.create(new FileAuditRecord(FileType.CDR_SUMMARY_FILE, file, error, null, null));
+        throw new IllegalStateException(error);
+    }
+
+
+
+    private CdrParseResult parseSummaryCdrs(String dir, String fileName, String checksum) {
```

Comment:
I'm a little worried (or paranoid) that this method might blow up the memory on the machine while processing a few hundred thousand records from the file. The list object will most likely end up in the large object heap allocation and we might run into problems with the GC and memory fragmentation.

Assuming I understand the design correctly, we take a file from a full day of cdrs and load it up in a memory list object. Then, we iterate through the list and fire event to process the CDRs. Would it be more efficient to fire the event as you go about parsing the file?